manticore/core/smtlib/expression.py

```diff
@@ -18,12 +18,20 @@ class ExpressionException(SmtlibError):
 class Expression:
     """ Abstract taintable Expression. """
 
-    __slots__ = ["_taint"]
+    # Static data bad, I know. Since the creation order is only used to order the expressions
+    # consistently, we don't need to worry about race conditions here. Collisions will introduce
+    # a little bit of nondeterminism, but should be rare. UNLESS state merging gets involved,
+    # in which case I haven't thought enough about what might happen.
+    _nth: int = 0  #: Total number of existing Expressions
```

Comment:
Though I think it's ultimately fine (provided no one else tries to use this field), this is an ugly solution and we can discuss potential alternatives. As-is, it's much cheaper than grabbing a timestamp or using a lock to ensure consistency.